### PR TITLE
Added legacy lookup export

### PR DESCRIPTION
**Pull request by abouffard**
Added the ability to use the console application to export a CSV
containing a mapping between 2525C symbology elements and their closest
2525D equivalents.  This can be useful to implementers who want to use
the latest 2525D symbology elements to create 2525C symbols.

### CODE REVIEW

**Comment by csmoore** [comment]
Linking to https://github.com/Esri/military-features-data/issues/190 since it appears that the changes to  `Military-All-Icons.csv + Military-Frame-And-Amplifier-Source-Icons.csv` in this commit may be related to that issue.

**Comment by abouffard** [comment]
@csmoore - please feel free to review this.  I am happy with it as is but I am always open to suggestions (I may not always act on them, but an extra set of eyes is always a good thing)..

File: source/JointMilitarySymbologyLibraryCS/librarian.cs [hunk 1]
```diff
@@ -852,55 +852,57 @@ internal LibraryAffiliation AffiliationByLegacyCode(string legacyStandardIdentit
             return retObj;
         }
 
-        internal LibraryAffiliation AffiliationByLegacyCode(string legacyStandardIdentityCode, string legacyDimensionCode, string legacyFirstLetterInFunction, string legacyCodingSchemeLetter)
-        {
-            LibraryAffiliation retObj = null;
+        // The following has been commented out as being potentially superflous
```
Comment (csmoore) [inline source/JointMilitarySymbologyLibraryCS/librarian.cs]: Thanks for clarifying commented out code.

**Comment by csmoore** [comment]
Generally looks good to me - I can't claim to understand the code. Output table looks as expected with 2 small things that recently changed but should be corrected prior to merge.

1. It appears that the updated "All_ID_Mapping_C_to_D.csv" assumes that the land unit special entity symbols will have a "Full Frame" central icon - this is not always the case and will cause the renderer not to find the correct icon. For example: S-G-UUMSET is erroneously marked "Full Frame" (and many other cases). This may be the result of some confusion of how I use the "FullFrame=true" but I use this exclusively for the central icon, because of cases like this.
2. The change you recently made, ex. `S-P-V-----,05000000` maps these retired symbols an invalid icon (which will hurt performance because of a query for non-existent icon) - if an icon does not exist no entry would work better here

My only other comments are 

1. Please update our style creation process to work with the new name you have chosen: "All_ID_Mapping_C_to_D.csv" 
2. The ids documentation for this file should have something similar to what we have for [the icon files](https://github.com/Esri/joint-military-symbology-xml/tree/master/svg) - we have an internal readme for this but it would be helpful to have this readme with the .csv for the user of this file (otherwise it would be quite cryptic)

Also going forward - we should try to make sure any changes to the icon files (in this PR - `Military-All-Icons.csv` - `Military-Frame-And-Amplifier-Source-Icons.csv`) have their own commit and reference the issue that prompted the change so we can better track the changes to these (since we now know this will break things)

**Comment by abouffard** [comment]
@csmoore , thanks for the review and comments.  I concur with all your suggestions and will take steps to address them.  I'll address your first two points above prior to merge and will handle the other two next sprint.

**Comment by abouffard** [comment]
Merged following changes made in response to @csmoore first two comments above.

**Pull request merged**